### PR TITLE
Clang fixes

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: folly/String.h [hunk 1]
```diff
@@ -185,9 +185,12 @@ std::string& stringAppendf(
  * As with vsnprintf() itself, the value of ap is undefined after the call.
  * These functions do not call va_end() on ap.
  */
-std::string stringVPrintf(const char* format, va_list ap);
-void stringVPrintf(std::string* out, const char* format, va_list ap);
-std::string& stringVAppendf(std::string* out, const char* format, va_list ap);
+std::string stringVPrintf(const char* format, va_list ap)
```
Comment (yfeldblum) [inline folly/String.h]: General style - attributes should precede the signature. Here and elsewhere.

File: folly/ClockGettimeWrappers.cpp [hunk 2]
```diff
@@ -23,7 +23,6 @@
 #include <time.h>
 
 #ifndef _WIN32
-#define _GNU_SOURCE 1
```
Comment (yfeldblum) [inline folly/ClockGettimeWrappers.cpp]: Is this known not to be needed? Here and elsewhere.

Comment (neheb) [reply folly/ClockGettimeWrappers.cpp]: _GNU_SOURCE is already defined here: https://github.com/facebook/folly/blob/master/CMake/FollyCompilerUnix.cmake#L23

I removed as it throws a redefinition error. Defining _GNU_SOURCE to 1 is wrong anyway.

File: folly/String.h (hunk 1 above)
Comment (neheb) [reply folly/String.h]: I only matched the way it already was.

**Comment by neheb** [comment]
I think this is what was meant by the first comment.

**Comment by neheb** [comment]
Rebased and added a bunch more. Please review.

**Pull request closed without merging**